src/commands/register.js

```diff
@@ -0,0 +1,257 @@
+                ctx.coreService = await startCore(preset, { wallet: true });
+
+                process.on('SIGINT', async () => {
+                  await ctx.coreService.stop();
+                  process.exit();
+                });
+              },
+            },
+            {
+              title: 'Import private key',
+              task: async (ctx) => importPrivateKey(ctx.coreService, ctx.fundSourcePrivateKey),
+            },
+            {
+              title: 'Sync Core with network',
+              enabled: () => preset !== PRESETS.LOCAL,
+              task: async (ctx) => waitForCoreSync(ctx.coreService),
+            },
+            {
+              title: 'Check balance',
+              task: async (ctx) => {
+                const balance = await getAddressBalance(ctx.coreService, ctx.fundSourceAddress);
+                if (balance <= MASTERNODE_DASH_AMOUNT) {
+                  throw new Error('You need to have more than 1000 Dash on your funding address');
+                }
+              },
+            },
+            {
+              title: 'Generate BLS keys',
+              task: async (ctx, task) => {
+                ctx.bls = await generateBlsKeys();
+
+                // eslint-disable-next-line no-param-reassign
+                task.output = `Public key: ${ctx.bls.publicKey}\nPrivate key: ${ctx.bls.privateKey}`;
+              },
+            },
+            {
+              title: 'Generate collateral address',
+              task: async (ctx, task) => {
+                ctx.collateral = await createNewAddress(ctx.coreService);
+
+                // eslint-disable-next-line no-param-reassign
+                task.output = `Address: ${ctx.collateral.address}\nPrivate key: ${ctx.collateral.privateKey}`;
+              },
+            },
+            {
+              title: 'Generate owner addresses',
+              task: async (ctx, task) => {
+                ctx.owner = await createNewAddress(ctx.coreService);
+
+                // eslint-disable-next-line no-param-reassign
+                task.output = `Address: ${ctx.owner.address}\nPrivate key: ${ctx.owner.privateKey}`;
+              },
+            },
+            {
+              title: 'Send 1000 Dash to collateral address',
+              task: async (ctx, task) => {
+                ctx.tx = await sendToAddress(
+                  ctx.coreService,
+                  ctx.fundSourcePrivateKey,
+                  ctx.fundSourceAddress,
+                  ctx.collateral.address,
+                  1000,
+                );
+
+                // eslint-disable-next-line no-param-reassign
+                task.output = `Hash: ${ctx.tx}`;
+              },
+            },
+            {
+              title: 'Wait for 15 confirmations',
+              enabled: () => preset !== PRESETS.LOCAL,
+              task: async (ctx) => (
+                new Observable(async (observer) => {
+                  await waitForConfirmations(
```

Comment:
Could we reuse `waitForBlocks` here?

Reply:
no, because your transaction could be included 5 blocks later, for example